src/openrct2/world/TileInspector.cpp

```diff
@@ -474,20 +474,39 @@ GameActionResult::Ptr tile_inspector_sort_elements_at(const CoordsXY& loc, bool
     return std::make_unique<GameActionResult>();
 }
 
+static GameActionResult::Ptr ValidateTileHeight(TileElement* const tileElement, int8_t heightOffset)
+{
+    int16_t newBaseHeight = static_cast<int16_t>(tileElement->base_height + heightOffset);
+    int16_t newClearanceHeight = static_cast<int16_t>(tileElement->clearance_height + heightOffset);
+    if (newBaseHeight < 0)
+    {
+        return std::make_unique<GameActionResult>(GA_ERROR::TOO_LOW, STR_CANT_LOWER_ELEMENT_HERE, STR_TOO_LOW);
+    }
+    else if (newBaseHeight > MAX_ELEMENT_HEIGHT)
+    {
+        return std::make_unique<GameActionResult>(GA_ERROR::TOO_HIGH, STR_CANT_RAISE_ELEMENT_HERE, STR_TOO_HIGH);
+    }
+    else if (newClearanceHeight < 0)
+    {
+        return std::make_unique<GameActionResult>(GA_ERROR::NO_CLEARANCE, STR_CANT_LOWER_ELEMENT_HERE, STR_NO_CLEARANCE);
+    }
```

Comment:
I can't actually see why we would ever fall here, but I didn't want to change the previous behavior

Reply:
This may be hit in case an element has a clearance height that is lower than its base height (who knows someone might have found a use of such a hack, I wouldn't be too surprised anymore).

Reply:
But would anything have a loww clearance*ok it's own" (without hacks)?

Reply:
Without hacks that's not possible indeed, then it would error out on the base height check already.

Reply:
Is it perhaps possible with saved track designs?

Reply:
I mean, I'm handling it anyway, just wanted to know if it was really possible